File: algo-expert/python/algo_expert/strings/easy/run_length_encoding.py
def run_length_encoding(string: str) -> str:
    encoded = []
    character = ""
    counter = 0
    for c in string:
        if c != character or counter == 9:
            if counter > 0:
                encoded.append(f"{counter}{character}")
            character = c
            counter = 1
        else:
            counter += 1
    encoded.append(f"{counter}{character}")
    return "".join(encoded)

File: algo-expert/typescript/src/strings/easy/RunLengthEncoding.ts
export function runLengthEncoding(string: string) {
  const chars = string.split('');
  let encodedChars = [];
  let counter = 1;
  let lastChar = chars[0];
  for (let i = 1; i < string.length; i++) {
    if (chars[i] != lastChar || counter == 9) {
      encodedChars.push(counter);
      encodedChars.push(lastChar);
      counter = 1;
      lastChar = chars[i];
    } else {
      counter++;
    }
  }
  encodedChars.push(counter);
  encodedChars.push(lastChar);
  return encodedChars.join('');
}
